File: api/models/general_settings.py
"""General (app-wide) admin settings models."""

from __future__ import annotations

from pydantic import BaseModel


class GeneralSettings(BaseModel):
    """Admin-editable app-wide settings."""

    # Public base URL the app is reachable at (scheme + host, no trailing slash),
    # used e.g. to build the API URL workers should poll. This is the admin
    # override (DB); ``None`` means unset, in which case the config default
    # (``config_public_base_url``) applies, then the request origin.
    public_base_url: str | None = None
    # Read-only default from server config (PUBLIC_BASE_URL env / config.yaml),
    # shown in the UI so admins can see what applies when no override is set.
    config_public_base_url: str | None = None


class GeneralSettingsUpdate(BaseModel):
    """Request model for updating general settings."""

    public_base_url: str | None = None

File: web/src/api/general-settings.ts
import { apiUrl, httpClient } from "./client.ts";

export interface GeneralSettings {
  public_base_url: string | null;
  // Read-only default from server config (PUBLIC_BASE_URL / config.yaml).
  config_public_base_url: string | null;
}

export const generalSettingsApi = {
  get: async (): Promise<GeneralSettings> => {
    const { json } = await httpClient(`${apiUrl}/general-settings`);
    return json;
  },

  update: async (data: { public_base_url: string | null }): Promise<GeneralSettings> => {
    const { json } = await httpClient(`${apiUrl}/general-settings`, {
      method: "PUT",
      body: JSON.stringify(data),
    });
    return json;
  },
};
